### ISSUE TITLE
PDF conversion with XeLaTeX fails on todo list

### ISSUE THREAD

**Issue by nuntius35** [jgm/pandoc#6825]
Input:
> test.md
> ````
> * [ ] item 
> ````

Command that fails:
````
$ pandoc test.md -o test.pdf --pdf-engine=xelatex
Error producing PDF.
! Undefined control sequence.
<recently read> \square
 
l.58 \item[$\square$]
````

Note that `pandoc test.md -o test.pdf` gives the desired output.

The fault seems to be that `amssymb` is not loaded, I obtain
````
$ pandoc test.md -s -o test.tex
$ less test.tex
[...]
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
  \usepackage{amssymb}
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
[...]
````

Version:
````
$ pandoc --version
pandoc.exe 2.11.1.1
Compiled with pandoc-types 1.22, texmath 0.12.0.3, skylighting 0.10.0.3,
citeproc 0.1.1.1, ipynb 0.1
User data directory: C:\Users\user\AppData\Roaming\pandoc
Copyright (C) 2006-2020 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
````
Could this be related to PR #6762? I don't know what the `$if(mathspec)$`-part is supposed to do.

**Comment by mb21** [comment]
what xelatex version do you have installed?

**Comment by nuntius35** [comment]
````
$ xelatex --version
MiKTeX-XeTeX 4.0.1 (MiKTeX 20.11)
© 1994-2008 SIL International, © 2009-2012 Jonathan Kew, © 2010-2012 Hàn Thế Thành, © 2012-2013 Khaled Hosny
TeX is a trademark of the American Mathematical Society
using bzip2 version 1.0.8, 13-Jul-2019
compiled with curl version 7.72.0; using libcurl/7.72.0 Schannel
compiled with expat version 2.2.10; using expat_2.2.10
compiled with fontconfig version 2.13.1; using 2.13.1
compiled with freetype2 version 2.10.3; using 2.10.3
compiled with graphite2 version 1.3.14; using 1.3.14
compiled with harfbuzz version 2.7.2; using 2.7.2
compiled with icu version 60.1; using 60.1
compiled with jpeg version 9.4
compiled with liblzma version 50020052; using 50020052
compiled with libpng version 1.6.37; using 1.6.37
compiled with libressl version LibreSSL 3.1.4; using LibreSSL 3.1.4
compiled with MiKTeX Application Framework version 4.0; using 4.0
compiled with MiKTeX Core version 4.2; using 4.2
compiled with MiKTeX Archive Extractor version 4.0; using 4.0
compiled with MiKTeX Package Manager version 4.0; using 4.0
compiled with poppler version 20.10.0
using teckit version 2.4
compiled with uriparser version 0.9.4
compiled with zlib version 1.2.11; using 1.2.11
````

**Comment by jgm** [comment]
This appears to be a regression.
See commit 83b54bb0b375c041fc59f9c0599c626582184903
(issue #6469) which removed amssymb for xelatex.
We were assured it wasn't necessary, but I guess that is wrong!

**Comment by jgm** [comment]
I guess, instead of using `\square` and `\boxtimes` in the todo lists, we could define macros `\UncheckedBox` and `\CheckedBox` (not math mode).  For pdflatex we could continue to use the amssymb symbols to define these, and for xelatex we could use a unicode character.  However, we might run into problems if the glyph isn't present in the font used.

**Comment by benabel** [comment]
The same ERROR appears with lualatex(This is LuaHBTeX, Version 1.12.0 (TeX Live 2020/Arch Linux)).

Loading `amssymb` fix the ERROR.

**Comment by benabel** [comment]
With Latin-modern I used `\newcommand\square{□}` as a temporary fix.

Here are the unicode dec and hex codes:

□ | 9633 | 25A1 |   | WHITE SQUARE

**Comment by jgm** [comment]
I'm gonna put amssymb back in the template, and make sure to load it before unicode-math.  In my tests this works fine with all three latex engines. It also works fine with the case that prompted #6762.

**Comment by jgm** [comment]
This should be closed by ea8097e41249387613cae2552866f1ea1cea1462
